chainimport/headers_import.go

```diff
@@ -0,0 +1,1816 @@
+
+	// Get chain tips for target stores.
+	_, bTipHeight, err := s.options.TargetBlockHeaderStore.ChainTip()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get target block header "+
+			"store chain tip: %v", err)
+	}
+	_, fTipHeight, err := s.options.TargetFilterHeaderStore.ChainTip()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get target filter header "+
+			"store chain tip: %v", err)
+	}
+	effectiveTipHeight := min(bTipHeight, fTipHeight)
+
+	// Create regions construct.
+	regions := &ProcessingRegions{
+		ImportStartHeight: importStartHeight,
+		ImportEndHeight:   importEndHeight,
+		EffectiveTip:      effectiveTipHeight,
+	}
+
+	// 1. Overlap region.
+	// This region contains headers that exist in both the import source and
+	// target stores, from the start of the import range up to the effective
+	// tip height.
+	overlapStart := importStartHeight
+	overlapEnd := min(effectiveTipHeight, importEndHeight)
+	regions.Overlap = HeaderRegion{
+		Start:  overlapStart,
+		End:    overlapEnd,
+		Exists: overlapStart <= overlapEnd,
+	}
+
+	// 2. Divergence region.
+	// This region contains headers where one store extends beyond the
+	// effective tip but still within the import range. It represents
+	// heights where targetblock and filter headers are out of sync and need
+	// reconciliation.
+	divergeStart := effectiveTipHeight + 1
+	divergeEnd := min(max(bTipHeight, fTipHeight), importEndHeight)
+	regions.Divergence = HeaderRegion{
+		Start:  divergeStart,
+		End:    divergeEnd,
+		Exists: bTipHeight != fTipHeight && divergeStart <= divergeEnd,
+	}
+
+	// 3. New Headers region.
+	// This region contains headers that are in the import source but not
+	// yet in either target store. They start one height beyond the highest
+	// tip of either store (ensuring no overlap with divergence region) and
+	// extend to the end of the import data. These headers need to be added
+	// to both stores. It only exists if there are headers beyond both tips.
+	//
+	// Note: This region is supposed to be processed after handling the
+	// overlap, and divergence regions, ensuring that any potential
+	// inconsistencies in existing data are resolved before adding new
+	// headers. This sequential processing guarantees that new headers
+	// are only added on top of a verified and consistent chain state.
+	newStart := max(bTipHeight, fTipHeight) + 1
+	newEnd := importEndHeight
+	regions.NewHeaders = HeaderRegion{
+		Start:  newStart,
+		End:    newEnd,
+		Exists: newStart <= newEnd,
+	}
+
+	return regions, nil
+}
+
+// verifyHeadersAtTargetHeight ensures headers at the specified height match
+// exactly between import and target sources by performing a byte-level
+// comparison.
+//
+// The function retrieves the header from both sources at the given height and
+// verifies they are identical, returning an error if any discrepancy is found.
+func (s *HeadersImport) verifyHeadersAtTargetHeight(height uint32) error {
```

Comment:
So this is part of a consistency check at the very end? Can we just handle this on restart instead?